# For the purpose of the 3.1 release, instead of sharing the binary and ooxml
# export, we introduce a separate sc/source/filter/xlsx that creates libxlsx -
# a filter for the XLSX files.  After the release, the changes will be
# reverted so that the binary and ooxml are shared again.
# The changes include:
#   sc/source/filter/excel/* -> sc/source/filter/excel/*
diff --git sc/source/filter/excel/xetable.cxx sc/source/filter/excel/xetable.cxx
index 3c2b1a5..8dea9e7 100644
--- sc/source/filter/excel/xetable.cxx
+++ sc/source/filter/excel/xetable.cxx
@@ -894,7 +894,7 @@ XclExpFormulaCell::XclExpFormulaCell(
         mxAddRec = rShrfmlaBfr.CreateOrExtendShrfmla( rScTokArr, aScPos );
 
     // no shared formula found - create a simple cell formula
-    if( !mxAddRec )
+    if( !mxAddRec || ! mxTokArr )
         mxTokArr = rRoot.GetFormulaCompiler().CreateFormula( EXC_FMLATYPE_CELL, rScTokArr, &aScPos );
 }
 
